src/src/Helpers/Networking.cpp

```diff
@@ -1,2051 +1,2184 @@
+      // => gets the value of field 1 at (or the last entry before) 23:59:00 of the channel 1637928
+      // All channels: "sendtohttp,api.thingspeak.com,80,channels/1637928/feeds.csv?end=2024-01-01%2023:59:00&results=1"
+      // => gets the value of each field of the channel 1637928 at (or the last entry before) 23:59:00 
+      //-----------------------------------------------------------------------------------------------------------------------------
+
+    if (httpCode == 200 && equals(host, F("api.thingspeak.com")) && (uri.endsWith(F("/last.csv")) || (uri.indexOf(F("results=1")) >= 0 && uri.indexOf(F(".csv")) >= 0))){
+      String result = http.getString();
+      result.replace(' ', '_'); // if using a single field with a certain time, the result contains a space and would break the code
+      const int posTimestamp = result.lastIndexOf(':');
+      if (posTimestamp >= 0){
+        result = parseStringToEndKeepCase(result.substring(posTimestamp), 3);
+        if (uri.indexOf(F("fields")) >= 0){                                                                           // when there is a single field call add the field number before the value
+          result = parseStringKeepCase(uri, 4, '/').substring(0, 1) + "," + result; // since the field number is always the fourth part of the url and is always a single digit, we can use this to extact the fieldnumber
+        }
+        eventQueue.addMove(strformat(
+            F("ThingspeakReply=%s,%s"),
+            parseStringKeepCase(uri, 2, '/').c_str(),
+            result.c_str()));
+      }
+    }
+    #endif
+
+    # if FEATURE_OMETEO_EVENT
+
+    // Generate an event with the response of an open-meteo request (https://open-meteo.com/en/docs)
+    // Example command:
+    // sendtohttp,api.open-meteo.com,80,/v1/forecast?latitude=52.52&longitude=13.41&current=temperature_2m,weather_code,wind_speed_10m&daily=temperature_2m_max,temperature_2m_min&forecast_days=1
+    // No need for an api key and it is free (daily requests are limited to 10,000 in the free version)
+    // Visit the URL and build your personal URL by selecting the location and values you want to receive.
+    // Supported variable kinds are current, hourly, daily!
+    // In rules you can grep the reply by the kind of weather variables with "On Openmeteo#<type> Do ..."
+    // e.g. "On Openmeteo#current Do ..."
+    // Note: hourly and daily results are arrays which can become very long.
+    // Best to make seperate calls. Especially for hourly results.
+
+    if ((httpCode == 200) && equals(host, F("api.open-meteo.com")))
+    {
+      // Length of the response is limited to 5000 characters
+      // ToDo: What would be right size?
+      if (uri.length() > 5000) {
+        addLog(LOG_LEVEL_ERROR, F("Response exceeds 5000 characters"));
+      }
+      else {
+        String str = http.getString();
+
+        // Process URL to get unique keys
+
+
+        auto getCombinedParams = [](String url, String paramName) {
```

Comment:
the `url` argument should be `const String& url`
Otherwise for each call the `url` String will be copied which is not needed.

I think you can also do the same with `const String& paramName`